### PR TITLE
[encoding] Tests for BOM detection.

### PR DESCRIPTION
**Pull request by andreubotella**
This change tests for whatwg/html#5359, but since I didn't see any tests for whether the BOM influences the decoding at all, I added one as well.

### CODE REVIEW

**Review by inexorabletash — APPROVED** [review]
LGTM, thanks!

**Review by inexorabletash — COMMENTED** [review]
Maybe to be completely pedantic have the comment specify that it starts with a UTF-8 BOM. 

<strike>(Also.... is a comment before the doctype declaration valid? Not sure offhand...)</strike>

File: encoding/bom-handling.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+);
+
+test(
+  () => {
+    assert_equals(document.characterSet.toLowerCase(), "utf-8");
+  },
+  "document.characterSet should match the BOM"
+);
+
+</script>
```
Comment (annevk) [inline encoding/bom-handling.html]:
```suggestion
</script>

```

File: encoding/bom-handling.html.headers [hunk 2]
```diff
@@ -0,0 +1 @@
+Content-Type: text/html; charset=windows-1252
```
Comment (annevk) [inline encoding/bom-handling.html.headers]:
```suggestion
Content-Type: text/html; charset=windows-1252

```

**Review by annevk — COMMENTED** [review]
Not sure if this works, but I'd prefer newlines at the end of files.

**Comment by annevk** [comment]
Thanks! I pinged @hsivonen since it seems somewhat suspect there would not be coverage for this. So would like to wait for that a bit, but do let me know if it's taking too long.

**Comment by hsivonen** [comment]
I believe we do have a separate test for testing that UTF-8 isn't heuristically detected. Still, I think this test would be more reliable if the only non-ASCII bytes were the BOM. I suggest removing the Japanese text from the test.

If you want to actually test decoding in addition to what `document.characterSet` reports, I suggest testing, in a separate test file, one (and only one) of the German sequences that [ced deliberately doesn't score towards UTF-8](https://github.com/google/compact_enc_det/blob/3e65e9dc136bac03e60457fdb213f13cf169905f/compact_enc_det/compact_enc_det.cc#L83-L92).

**Comment by hsivonen** [comment]
(Of course, if the BOM was ignored, the `meta` should take precedence over heuristic sniffing, but let's still avoid making the test too easy to pass for the wrong reason.)

**Comment by hsivonen** [comment]
> it seems somewhat suspect there would not be coverage for this

It indeed seem the `encoding/` directory does not have a test for this. The existing BOM tests are for checking that the right number of BOMs is removed.

**Comment by andreubotella** [comment]
Since this test is for whatwg/html#5359, which fixes the long-standing spec bug where the document's character encoding doesn't take the BOM into account, I thought it made sense to test both the decoding and `document.characterSet`. But I guess the chances that some implementation will get the document's character encoding right but the actual decoding wrong are so low that there's no need to test for detection here.

**Comment by hsivonen** [comment]
Looks good to me. Thanks!

**Pull request merged**